src/Microsoft.Health.Fhir.Api.UnitTests/Controllers/AadSmartOnFhirProxyControllerTests.cs

```diff
@@ -0,0 +1,118 @@
+        private readonly FeatureConfiguration _featureConfiguration = new FeatureConfiguration();
+        private readonly AadSmartOnFhirProxyController _controller;
+
+        public AadSmartOnFhirProxyControllerTests()
+        {
+            _securityConfiguration.EnableAadSmartOnFhirProxy = true;
+            _securityConfiguration.Enabled = true;
+            _securityConfiguration.Authorization = new AuthorizationConfiguration
+            {
+                Enabled = true,
+            };
+            _securityConfiguration.Authentication = new AuthenticationConfiguration
+            {
+                Audience = "testaudience",
+                Authority = "http://testauthority.com/v2.0",
+            };
+
+            string openIdConfiguration;
+            using (StreamReader r = new StreamReader(Assembly.GetExecutingAssembly().
+                GetManifestResourceStream("Microsoft.Health.Fhir.Api.UnitTests.Controllers.openid-configuration.json")))
+            {
+                    openIdConfiguration = r.ReadToEnd();
+            }
+
+            var httpMessageHandler = new TestHttpMessageHandler(new HttpResponseMessage()
+            {
+                StatusCode = System.Net.HttpStatusCode.OK,
+                Content = new StringContent(openIdConfiguration),
+            });
+
+            var httpClient = new HttpClient(httpMessageHandler);
+
+            _httpClientFactory.CreateClient().Returns(httpClient);
+
+            _controller = new AadSmartOnFhirProxyController(
+                Options.Create(_securityConfiguration),
+                _httpClientFactory,
+                _urlResolver,
+                _logger);
+        }
+
+        [Theory]
+        [InlineData(null, null, null, null, null, null, null)]
+        [InlineData("code", null, null, null, null, null, null)]
+        [InlineData("code", "clientId", null, null, null, null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", null, null, null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", "launch", null, null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", "launch", "scope", null, null)]
+        [InlineData("code", "clientId", "https://testurl.com", "launch", "scope", "state", null)]
+
+        public void GivenIncompleteQueryPamas_WhenAuthorizeRequestAction_ThenRedirectResultReturned(
+            string responseType, string clientId, string redirectUri, string launch, string scope, string state, string aud)
+        {
+            Uri redirect = null;
+            if (!string.IsNullOrEmpty(redirectUri))
+            {
+                redirect = new Uri(redirectUri);
+            }
+
+            _urlResolver.ResolveRouteNameUrl(Arg.Any<string>(), Arg.Any<IDictionary<string, object>>()).Returns(redirect);
+
+            var result = _controller.Authorize(responseType, clientId, redirect, launch, scope, state, aud);
+
+            Assert.IsType<RedirectResult>(result);
```

Comment:
Could possibly do an `as RedirectResult` and just assert isn't null. That way you don't have to do the cast below.